code/modules/power/singularity/emitter.dm

```diff
@@ -186,36 +190,59 @@
 		add_load(active_power_usage)
 		fire_beam()
 
-/obj/machinery/power/emitter/proc/fire_beam()
-	src.last_shot = world.time
-	if(src.shot_number < 3)
-		src.fire_delay = 20
-		src.shot_number ++
-	else
-		src.fire_delay = rand(minimum_fire_delay,maximum_fire_delay)
-		src.shot_number = 0
-	var/obj/item/projectile/A = new projectile_type(src.loc)
-	A.setDir(src.dir)
-	playsound(src.loc, projectile_sound, 25, 1)
+/obj/machinery/power/emitter/proc/fire_beam(atom/targeted_atom, mob/user)
+	var/turf/targets_from = get_turf(src)
+	var/obj/item/projectile/P = new projectile_type(targets_from)
+	playsound(src.loc, projectile_sound, 50, 1)
 	if(prob(35))
 		var/datum/effect_system/spark_spread/s = new /datum/effect_system/spark_spread
 		s.set_up(5, 1, src)
 		s.start()
 	switch(dir)
 		if(NORTH)
-			A.yo = 20
-			A.xo = 0
+			P.yo = 20
+			P.xo = 0
 		if(EAST)
-			A.yo = 0
-			A.xo = 20
+			P.yo = 0
+			P.xo = 20
 		if(WEST)
-			A.yo = 0
-			A.xo = -20
+			P.yo = 0
+			P.xo = -20
 		else // Any other
-			A.yo = -20
-			A.xo = 0
-	A.starting = loc
-	A.fire()
+			P.yo = -20
+			P.xo = 0
+	if(!manual)
+		src.last_shot = world.time
+		if(src.shot_number < 3)
+			src.fire_delay = 20
+			src.shot_number ++

```

Comment:
A lot of these were there originally and are only "added" because I had to cut and paste a lot of code around, but im cutting them.